### PR TITLE
Update Matomo Java Tracker to 3.2.0

### PR DESCRIPTION
**Pull request by dheid**
### Hey @f1sh1918 and digitalfabrik team,

Just wanted to give your project a little boost with the latest and greatest from Matomo Java Tracker! 🚀

### Changes in a Nutshell:

- **Bug Fixes & Performance Boosts:** Say goodbye to some pesky bugs and enjoy a faster, smoother tracking experience.

- **Security Makeover:** We're getting the latest security swag to keep our project locked down.

- **Stay Cool with Compatibility:** Now we're best buddies with Matomo versions up to 5, so no more awkward compatibility issues.

- **Less Baggage:** Matomo Java Tracker 3.2.0 is lean and mean with fewer dependencies. Less stuff, more speed!

- **New Toys:** Check out the cool new dimension parameter for super-customizable tracking.

- **Java 11 Party:** We're upgrading to the latest Java 11 implementation. It's like giving our project a cool new set of wheels.

- **Smooth Moves:** If you're coming from version 2, no worries! There's a handy migration guide [here](https://github.com/matomo-org/matomo-java-tracker#migration-from-version-2-to-302) to make the switch painless.

### How to Test:

1. Update to Matomo Java Tracker version 3.2.0.
2. Make sure our project still builds like a champ.
3. Double-check that our Matomo tracking mojo is still on point.
4. Test out any new features and give that dimension parameter a spin.

### Where to Get Help:

- Dive into the [Javadoc](https://javadoc.io/doc/org.piwik.java.tracking/matomo-java-tracker/3.2.0/index.html) for the nitty-gritty details.
- Got issues or bright ideas? Hit up the [Matomo Java Tracker GitHub](https://github.com/matomo-org/matomo-java-tracker).

### Your Feedback Matters:

Give it a whirl and let us know how it goes! Your feedback is like gold to us. 🌟

Cheers,
Daniel

### CODE REVIEW

**Comment by f1sh1918** [comment]
thx @dheid for your contribution :)
We gonna check your pr and put it in our next major release

**Comment by dheid** [comment]
@f1sh1918 That's great! Thank you very much!

**Comment by dheid** [comment]
@f1sh1918 ping

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/matomo/Matomo.kt [hunk 1]
```diff
@@ -153,7 +150,7 @@ object Matomo {
         if (params.searchText === null && params.categoryIds === null) return
         sendTrackingRequest(
             projectConfig.matomo,
-            MatomoRequest.builder()
+            MatomoRequest.request()
                 .actionName(query)
                 .searchQuery((params.searchText ?: "") + "?categories=" + (params.categoryIds?.joinToString(",") ?: ""))
```
Comment (sarahsporck) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/matomo/Matomo.kt:155]: Could you also split search query and search category, as this works now as intended? Would be awesome :)
```suggestion
     .searchCategory(params.categoryIds?.joinToString(","))
     .searchQuery(params.searchText ?: "")
```

**Review by sarahsporck — CHANGES_REQUESTED** [review]
Really awesome! Thanks for this PR! I left one small comment. Besides that this is ready to go.

---

Tested with AppSearch & CardActivation.

**Comment by dheid** [comment]
You're welcome! I accepted your changes and increased the version to 3.2.0.

**Review by sarahsporck — APPROVED** [review]
Very nice! Thanks a lot.

**Pull request merged**